static/app/views/eventsV2/savedQuery/index.tsx

```diff
@@ -161,7 +161,7 @@ class SavedQueryButtonGroup extends React.PureComponent<Props, State> {
       (savedQuery: SavedQuery) => {
         const view = EventView.fromSavedQuery(savedQuery);
 
-        setBannerHidden(true);
+        Banner.dismiss('discover');
```

Comment:
Banner is now responsible for dismissing, so I added this helper method onto the Banner prototype.